GPU/Common/SplineCommon.cpp

```diff
@@ -223,6 +223,39 @@ static void spline_knot(int n, int type, float *knot) {
 	}
 }
 
+static void TessellateSplinePatchHardware(u8 *&dest, u16 *indices, int &count, const SplinePatchLocal &spatch) {
+	SimpleVertex *&vertices = (SimpleVertex*&)dest;
+
+	for (int tile_v = 0; tile_v < spatch.tess_v + 1; ++tile_v) {
+		for (int tile_u = 0; tile_u < spatch.tess_u + 1; ++tile_u) {
+			float u = ((float)tile_u / (float)spatch.tess_u);
+			float v = ((float)tile_v / (float)spatch.tess_v);
+
+			SimpleVertex &vert = vertices[tile_v * (spatch.tess_u + 1) + tile_u];
+
+			vert.pos.x = u;
+			vert.pos.y = v;
+
+			// For compute normal
+			vert.nrm.x = 1.0f / (float)spatch.tess_u;
```

Comment:
These divisions can be moved outside the loop. The compiler might do that for us, but it's far from guaranteed and divisions can be a little expensive on some hardware. Might even be able to use a uniform, but not sure if that would make a difference.